bal_tools/subgraph.py

```diff
@@ -30,6 +30,33 @@
     "fraxtal": "https://graph.data.aura.finance/subgraphs/name/aura-finance-fraxtal",
     "avalanche": "https://subgraph.satsuma-prod.com/cae76ab408ca/1xhub-ltd/aura-finance-avalanche/api",
 }
+# https://raw.githubusercontent.com/balancer/docs-v3/refs/heads/v3-outline/docs/data-and-analytics/data-and-analytics/subgraph.md
+VAULT_V3_SUBGRAPHS_BY_CHAIN = {
+    "arbitrum": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/Ad1cgTzScNmiDPSCeGYxgMU3YdRPrQXGkCZgpmPauauk",
+    "base": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/9b7UBHq8DXxrfGsYhAzF3jZn5mNRgZb5Ag18UL9GJ3cV",
+    "ethereum": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/4rixbLvpuBCwXTJSwyAzQgsLR8KprnyMfyCuXT8Fj5cd",
+    "gnosis": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/DDoABVc9xCRQwuXRq2QLZ6YLkjoFet74vnfncQDgJVo2",
+}
+VAULT_V3_SUBGRAPHS_BY_CHAIN_DEV = {
+    "arbitrum": "https://api.studio.thegraph.com/query/75376/balancer-v3-arbitrum/version/latest",
+    "base": "https://api.studio.thegraph.com/query/75376/balancer-v3-base/version/latest",
+    "mainnet": "https://api.studio.thegraph.com/query/75376/balancer-v3/version/latest",
+    "gnosis": "https://api.studio.thegraph.com/query/75376/balancer-v3-gnosis/version/latest",
+    "sepolia": "https://api.studio.thegraph.com/query/75376/balancer-v3-sepolia/version/latest",
+}
+POOLS_V3_SUBGRAPHS_BY_CHAIN = {
+    "arbitrum": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/EjSsjATNpZexLhozmDTe9kBHpZUt1GKjWdpZ2P9xmhsv",
+    "base": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/42QYdE4P8ZMKgPx4Mkw1Vnx3Zf6AEtWFVoeet1HZ4ntB",
+    "ethereum": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/C4tijcwi6nThKJYBmT5JaYK2As2kJGADs89AoQaCnYz7",
+    "gnosis": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/yeZGqiwNf3Lqpeo8XNHih83bk5Tbu4KvFwWVy3Dbus6",
+}
+POOLS_V3_SUBGRAPHS_BY_CHAIN_DEV = {
+    "arbitrum": "https://api.studio.thegraph.com/query/75376/balancer-pools-v3-arbitrum/version/latest",
+    "base": "https://api.studio.thegraph.com/query/75376/balancer-pools-v3-base/version/latest",
+    "mainnet": "https://api.studio.thegraph.com/query/75376/balancer-pools-v3/version/latest",
+    "gnosis": "https://api.studio.thegraph.com/query/75376/balancer-pools-v3-gnosis/version/latest",
+    "sepolia": "https://api.studio.thegraph.com/query/75376/balancer-pools-v3-sepolia/version/latest",
+}
```

Comment:
not sure if the intention is to keep these urls hardcoded, but if you wanted a programmatic way to scrape the urls from that subgraph.md file, there is actually a pretty elegant solution using `pd.read_html`

```py
dfs = pd.read_html("https://github.com/balancer/docs-v3/blob/v3-outline/docs/data-and-analytics/data-and-analytics/subgraph.md")

def url_dict_from_df(df):
    return (
        dict(zip(df['Network'], df['Production URL'])),
        dict(zip(df['Network'], df['Development URL (rate-limited)']))
    )

VAULT_V3_SUBGRAPHS_BY_CHAIN, VAULT_V3_SUBGRAPHS_BY_CHAIN_DEV = url_dict_from_df(dfs[1])
POOLS_V3_SUBGRAPHS_BY_CHAIN, POOLS_V3_SUBGRAPHS_BY_CHAIN_DEV = url_dict_from_df(dfs[2])
```

Reply:
lets give it a shot. i didnt want to spend too much time on programmatic retrieval, because im not sure how reliable the current structure is. but lets give it a shot, your solution is indeed pretty elegant and we can always revert back to the manual approach

[9096f38](https://github.com/BalancerMaxis/bal_tools/pull/83/commits/9096f381f4b9b5801bc1d270092a62e19e075910)